# Triton compiles their CUDA launcher at runtime in order to not need CUDA
# just to build the Triton library and compile modules. We need to compile it
# ahead of time in order to run the tests correctly.

--- a/third_party/nvidia/backend/driver.py
+++ b/third_party/nvidia/backend/driver.py
@@ -63,13 +63,8 @@
         return cls.instance
 
     def __init__(self):
-        mod = compile_module_from_src(
-            src=Path(os.path.join(dirname, "driver.c")).read_text(),
-            name="cuda_utils",
-            library_dirs=library_dirs(),
-            include_dirs=include_dirs,
-            libraries=libraries,
-        )
+        from ._C import cuda_utils
+        mod = cuda_utils
         global PyCUtensorMap
         global PyKernelArg
         global ARG_CONSTEXPR
